src/index.ts

```diff
@@ -1,12 +1,13 @@
 // https://github.com/libp2p/js-libp2p/issues/1269#issue-1280763716
-export class CodeError<T extends Record<string, any> = Record<string, never>> extends Error {
-  code: string
-  props: T
-
-  constructor (message: string, code: string, props?: T) {
+export class CodeError<T extends Readonly<Record<string, any>>> extends Error {
+  constructor (
+    readonly message: string,
+    readonly code: string,
+    readonly props: T = {} as T // eslint-disable-line @typescript-eslint/consistent-type-assertions
+  ) {
     super(message)
 
     this.code = code
```

Comment:
If you set `readonly code` in the constructor function signature, you don't need this line.

Reply:
right i forgot about this